### DISCUSSION TITLE
Machines with dead heartbeats don't get removed?

### DISCUSSION THREAD

**Discussion by nschloe** [keygen-sh/community#55] [Q&A] [answered]
I'm playing around with heartbeats. One thing I don't quite get: After a machines starts a heartbeat and fails to maintain it, its _Heartbeat Status_ is `DEAD`. However, it still appears in list of machines, and also takes a seat on the corresponding license.

Do I have to manually remove machines with dead heartbeats?

**Comment by nschloe** [comment, marked as answer]
Checking back now, and it appears to work again. Not sure what the reason was.

**Reply by ezekg** [reply]
Dead machines are not culled instantly — sometimes there is a slight delay. The rate at which a machine is culled will also depend on the heartbeat window, as well as the policy's heartbeat cull and resurrection strategies. For example, with a heartbeat window of 1 hour, and a cull strategy of `DEACTIVATE_DEAD` and a resurrection strategy of `15_MINUTE_REVIVE`, a "dead" machine won't be deactivated until at least 15 minutes after its time of death.

**Reply by nschloe** [reply]
All good to know, thanks!
